### PR TITLE
US129569 - Step 2 - Verify Destination Course Selection

### PR DESCRIPTION
**Pull request by Syeda-Azim**
https://rally1.rallydev.com/#/431372673576d/iterationstatus?detail=%2Fuserstory%2F602844268085

### CODE REVIEW

File: src/components/catoc-course-selection.js [hunk 1]
```diff
@@ -126,7 +126,8 @@ class CatocCourseSelection extends LocalizeMixin(LitElement) {
 		return html`
 			<d2l-list-item
 				key=${course.OrgUnit.Id}
-				selectable>
+				selectable
+				?selected="${this._selectedCourseIds.includes(course.OrgUnit.Id.toString())}">
```
Comment (Syeda-Azim) [inline src/components/catoc-course-selection.js:130]: This fixes the selected course issue on search. Now, the course checkbox will only be checked if it's selected.

File: src/components/catoc-summary.js [hunk 2]
```diff
@@ -85,11 +95,15 @@ class CopyAssignmentToOtherCoursesSummary extends LocalizeMixin(LitElement) {
 				<h4>${this._dropbox.Name}</h4>
 				<p class="title">${this._course.Name} (${this._course.Code})</p>
 				<div class="d2l-checkbox-container">
-					<d2l-input-checkbox id="rubric-checkbox">
+					<d2l-input-checkbox
+						id="rubric-checkbox"
+						?disabled=${rubrics.length === 0}
+						?checked=${this.inludeRubrics && rubrics.length !== 0}>
```
Comment (Syeda-Azim) [inline src/components/catoc-summary.js:101]: - Rubrics checkbox is diabled when there is no rubrics in the assignmet.
- Rubrics checkbox is checked by default.

File: src/components/catoc-summary.js [hunk 3]
```diff
@@ -85,11 +95,15 @@ class CopyAssignmentToOtherCoursesSummary extends LocalizeMixin(LitElement) {
...
 						${this.localize('rubricsCheckbox', { rubricsCount: rubrics.length })}
 					</d2l-input-checkbox>
 					${this._renderTooltip(rubrics)}
-					<d2l-input-checkbox>
+					<d2l-input-checkbox selected
+						?checked=${this.inludeGradeItems}>
```
Comment (Syeda-Azim) [inline src/components/catoc-summary.js:106]: Grade Items checkbox is checked by default.

**Review by lweber-d2l — APPROVED** [review]

File: copy-assignment-to-other-courses.js [hunk 4]
```diff
@@ -19,7 +20,17 @@ class CopyAssignmentToOtherCourses extends LocalizeMixin(LitElement) {
 			},
 			_selectedCourseIds: {
 				type: Array
+			},
+			_selectedCourses: {
```
Comment (jorge-ramirez-arredondo) [inline copy-assignment-to-other-courses.js:24]: Should `_selectedCourses` replace `_selectedCourseIds`? I assume `_selectedCourses` is an array of objects that each have their course id in there, so having another array here is just creating duplicate state to keep track of and make sure to update.

File: copy-assignment-to-other-courses.js [hunk 5]
```diff
@@ -68,23 +87,75 @@ class CopyAssignmentToOtherCourses extends LocalizeMixin(LitElement) {
...
-				</d2l-button>
-			</d2l-floating-buttons>
+			${this._renderSteps()}
+			${this._renderFloatingButtons()}
 		`;
 	}
 
+	_destinationCoursesChange(e) {
+		this._selectedCourses = e.detail.destinationCourses;
+		this.requestUpdate();
```
Comment (jorge-ramirez-arredondo) [inline copy-assignment-to-other-courses.js:97]: I don't think this `requestUpdate` call is needed. I think `destinationCourses` is a new array every time this event fires, so just setting `this._selectedCourses` should trigger an update.

**Pull request merged**